### PR TITLE
docs: destination docs for k8s

### PR DESCRIPTION
**Pull request by DanStough**
### Description
Adding Kubernetes documentation for the new transparent proxy Destination feature.

### Testing & Reproduction steps
N/A

### Links
Additional Doc PR #13950 

### PR Checklist

* [ ] ~updated test coverage~
* [X] external facing docs updated
* [X] not a security concern

### CODE REVIEW

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (dhiaayachi) [inline website/content/docs/k8s/connect/terminating-gateways.mdx:102]: We should at least include SNI with CAFiles.

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (dhiaayachi) [inline website/content/docs/k8s/connect/terminating-gateways.mdx:103]: For IP with no TLS we should say that you can encrypt traffic at the tgtw using SNI and certs.

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (dhiaayachi) [inline website/content/docs/k8s/connect/terminating-gateways.mdx:108]: Same here about encrypting traffic at the tgtwy

**Review by dhiaayachi — APPROVED** [review]
Few comments about SNI and certs.

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (DanStough) [reply website/content/docs/k8s/connect/terminating-gateways.mdx:102]: Per our discussion, let's add this to the config entry field descriptions.

**Review by dhiaayachi — APPROVED** [review]
🚢

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (trujillo-adam) [inline website/content/docs/k8s/connect/terminating-gateways.mdx:92]:
```suggestion
1. If [`TransparentProxy`](/docs/k8s/helm#v-connectinject-transparentproxy) is enabled, you can declare external endpoints in the [`Destination`](/docs/connect/config-entries/service-defaults#terminating-gateway-destination) field of `service-defaults`.
```
The docs for the `destination` field are still in production, correct?

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (trujillo-adam) [inline website/content/docs/k8s/connect/terminating-gateways.mdx:93]:
```suggestion
1. You can add the service as a node in the Consul catalog.
```
If we mean Kubernetes Node, then we should capitalize to distinguish between nodes in Consul and Nodes in Kubernetes.

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (trujillo-adam) [inline website/content/docs/k8s/connect/terminating-gateways.mdx:102]: A few things:
1. As far as I can see, we don't use the checkmark or cross-out icons anywhere else. For consistency, I think we should remove them.
2. It took me a while to figure out that these table represents a matrix with conditions along the x and y axises, as opposed to tabular reference information where the column headings represent a features associated with the items in the left-most column.
3. We introduce the matrix as describing limitations, but most of the reference points show successful calls. I would expect the matrix that describes limitations to only show instances when something is unsuccessful.

I think I would put this information into a more linear format.

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (trujillo-adam) [inline website/content/docs/k8s/connect/terminating-gateways.mdx:111]:
```suggestion
You provide a `caFile` to secure traffic between unencrypted clients that connect to external services through the terminating gateway.
Refer to [Create the configuration entry for the terminating gateway](/docs/k8s/connect/terminating-gateways#create-the-configuration-entry-for-the-terminating-gateway) for details.
```
This didn't seem like orthogonal information that should be labeled in a side note. 
Also restated in active voice.

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (trujillo-adam) [inline website/content/docs/k8s/connect/terminating-gateways.mdx:366]:
```suggestion
**For external services registered with the Consul catalog**
```
I get that we're bolding this sentence fragment so that it reads like a label for the following command, but I wonder if it wouldn't be better to wrap a CodeBlockConfig element around it and use the heading=""" attribute to make it look nicer.

**Review by trujillo-adam — COMMENTED** [review]
A couple nits and more-than-nits comments.

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (DanStough) [reply website/content/docs/k8s/connect/terminating-gateways.mdx:92]: Those docs are not merged but already approved here: https://github.com/hashicorp/consul/pull/13950 
I'll wait until those get merged before merging this one.

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (DanStough) [reply website/content/docs/k8s/connect/terminating-gateways.mdx:93]: Ah, TIL. 🙏

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (DanStough) [reply website/content/docs/k8s/connect/terminating-gateways.mdx:111]: I used the note more to emphasize "pay attention" rather than "something related", but happy to adopt the convention here.

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (DanStough) [reply website/content/docs/k8s/connect/terminating-gateways.mdx:366]: OK, I'll try that out.

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (DanStough) [reply website/content/docs/k8s/connect/terminating-gateways.mdx:102]: I was using the table [here](https://learn.hashicorp.com/tutorials/consul/terminating-gateways-connect-external-services#prerequisites) as prior art, although now I see they use the 🚫 symbols instead. I'm not married to this but I would like your take on the reference provided.

When you say linear format, would you be OK with something like the following that doesn't use the Y axis? I like some sort of table organization because I personally found it hard to grok all these possible scenarios without having them laid out together.

| Service Layer | TLS | Client dials | Works out of the Box | Notes |
|---|---|---|---|---|
| L4 | No | Hostname | 🚫 | Because the Sidecar is not protocol-aware, it will not be able to identify traffic going to the external service. |

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (DanStough) [reply website/content/docs/k8s/connect/terminating-gateways.mdx]: I agree it looks nicer!

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (DanStough) [reply website/content/docs/k8s/connect/terminating-gateways.mdx]: See changes for the actual layout I picked.

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (lkysow) [inline website/content/docs/k8s/connect/terminating-gateways.mdx:95]: non-blocking: It would be amazing if you folks could provide some examples. e.g. for `https://www.twitter.com` use this `ServiceDefaults` (I'm guessing with protocol: tcp), and you can then call Twitter from your service with `curl https://www.twitter.com`

Comment (DanStough) [reply website/content/docs/k8s/connect/terminating-gateways.mdx:95]: I think this doc reads a little like a tutorial that ends with a similar "try this" command. I would favor replacing `example.com` with something like `httpbin.org` with a simple response. WDYT @trujillo-adam @lkysow?

Comment (lkysow) [reply website/content/docs/k8s/connect/terminating-gateways.mdx:95]: Ahh I see, sorry I didn't look at the full context. I still think that reading the table of caveats is hard to understand when mapping to what my use-case is. I don't have strong feelings about replacing httpbin but I do think that having examples for when I should use L4 vs L7 and what the servicedefaults and eventual `curl` calls would look like would make it much easier to understand.

Comment (DanStough) [reply website/content/docs/k8s/connect/terminating-gateways.mdx:95]: I agree the table is challenging, but I struggled to find a better representation. The best solution would be to have some permissive or auto setting to make figuring this out more accessible, but that's forward-looking. Open to any suggestions here.

Comment (trujillo-adam) [reply website/content/docs/k8s/connect/terminating-gateways.mdx:95]: W/r/t examples, I'm in favor of using as realistic configurations as possible so that readers can form a mental model of what we are describing. Regarding the table of behaviors, I actually think this version of the table is pretty straightforward. There are a few blank cells under the Notes column that we should complete, but as an outside reader I feel that I would be able to align my use case to the information being presented. Maybe Notes should be changed to Description.

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (trujillo-adam) [inline website/content/docs/k8s/connect/terminating-gateways.mdx:109]: A couple of cells are empty in the Notes column. For consistency, we should describe the reason for the behavior for each row.

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (DanStough) [reply website/content/docs/k8s/connect/terminating-gateways.mdx:95]: OK, I think we'll keep it as-is for now and work on more examples as this feature develops. We have some additional features planned where we can add to this.

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (DanStough) [reply website/content/docs/k8s/connect/terminating-gateways.mdx]: Added some content. I also rebased.

File: website/content/docs/k8s/connect/terminating-gateways.mdx
Comment (trujillo-adam) [inline website/content/docs/k8s/connect/terminating-gateways.mdx:109]:
```suggestion
| L4 | Hostname | No | Not allowed | The sidecar is not protocol aware and can not identify traffic going to the external service. |
| L4 | IP | No | Allowed | There are no limitations on dialing IPs without TLS. |
| L7 | Hostname | Yes | Not allowed | Because traffic is already encrypted before the sidecar, it cannot route as L7 traffic. |
| L7 | IP | Yes | Not allowed | Because traffic is already encrypted before the sidecar, it cannot route as L7 traffic. |
| L7 | Hostname | No | Allowed | A `Host` or `:authority` header is required. |
| L7 | IP | No | Allowed | There are no limitations on dialing IPs without TLS.  |
```
Just a few super minor style things: use simple present tense and use lower case unless it's a proper name.

**Review by trujillo-adam — APPROVED** [review]
There are a few style things that we should fix, but approving so I don't slow you down.

**Pull request merged**
